tests/OmniSharp.Tests/DiagnosticsFacts.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OmniSharp.Models;
+using Xunit;
+
+namespace OmniSharp.Tests
+{
+    public class DiagnosticsFacts
+    {
+        [Fact]
+        public async Task CodeCheckSpecifiedFileOnly()
+        {
+            var workspace = TestHelpers.CreateSimpleWorkspace(new Dictionary<string, string>
+            {
+                { "a.cs", "class C { int n = true; }" }
+            });
+
+            var controller = new OmnisharpController(workspace, null);
+            var quickFixes = await controller.CodeCheck(new Request() { FileName = "a.cs" });
+
+            Assert.Equal(2, quickFixes.QuickFixes.Count());
```

Comment:
I was gonna take a look at this the other night, but never got around to it. 

I think that all the other code that relies on Equals and GetHashCode is covered by tests now so we should just be able to remove it and see.